jasmine-2.5.2/src/Game.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+function Game() {
+  this._frame = 1;
+  this._roll = 1;
+  this._pins = 10;
+}
+Game.prototype.getFrame = function() {return this._frame;};
+Game.prototype.getRoll = function() {return this._roll;};
+Game.prototype.getPins = function() {return this._pins;};
+
+Game.prototype.setPins = function(hits) {
+  this._pins -= hits;
+}
+
+Game.prototype.resetFrameRollAndPins = function() {
+  if (this.isEndOfFrame()) {
+    this.setUpNewFrame()
+  } else {
+    this.setUpNewRoll();
+  }
+}
+Game.prototype.isEndOfFrame = function() {
+  return (this.getRoll() == 2 || this.areNoPinsLeft());
+}
+Game.prototype.areNoPinsLeft = function() {
+  return this.getPins() == 0;
+}
+
+Game.prototype.setUpNewFrame = function() {
+  this._frame++;
+  this._pins = 10;
+  this._roll = 1;
+}
+Game.prototype.setUpNewRoll = function() {
+  this._roll = 2;
+}
+
+Game.prototype.isOver = function(extraRolls) {
+  return (this.getFrame() > 10 && extraRolls == 0);
+}
```

Comment:
Lovely short methods. Perhaps you could start eliminating the magic numbers by extracting them to constants. From there, a way of treating the scores might emerge (possibly using a [Quantity Pattern](http://wiki.c2.com/?QuantityPattern), but that's off the top of my head)